### PR TITLE
RATEST-242: RefApp 2.x Patient condition workflow test

### PR DESCRIPTION
**Pull request by mherman22**
**Ticket ID**
https://issues.openmrs.org/browse/RATEST-242

**reference**
https://github.com/openmrs/openmrs-contrib-qaframework/commit/247b49d8974a89f5f3e3d6c2c162639f3fec6349

Work Done

1. Added visibility Elements to the editInactive element and delete inactive element.
2. Otherwise this is a continuation of https://github.com/openmrs/openmrs-contrib-qaframework/pull/174

cc: @kdaud @sherrif10

### CODE REVIEW

**Comment by kdaud** [comment]
@mherman22 is the commit reflecting what is taking place in the PR?

File: src/test/java/org/openmrs/contrib/qaframework/automation/ConditionsSteps.java [hunk 1]
```diff
@@ -171,33 +181,36 @@ public void successiveInactiveEdition() {
...
-		conditionsPage.clickInActiveTab();
-		name = conditionsPage.getFirstConditionName();
+	}
+		
+	@Then("System should trash first inactive condition")
+		public void systemDeletesInActiveCondition() {
+		String name = conditionsPage.getFirstConditionName();
 		if (StringUtils.isNotBlank(name)) {
-			assertNull(driver.findElement(By.linkText(name)));
+			assertNull(driver.findElement(By.xpath(name)));
```
Comment (sherrif10) [inline src/test/java/org/openmrs/contrib/qaframework/automation/ConditionsSteps.java:213]: isnt there anything specific than asserting an element.

Comment (mherman22) [reply src/test/java/org/openmrs/contrib/qaframework/automation/ConditionsSteps.java:213]: not been able to find one. would you care to suggest?

File: src/test/java/org/openmrs/contrib/qaframework/automation/ConditionsSteps.java [hunk 2]
```diff
@@ -25,27 +26,30 @@
 import io.cucumber.java.en.Then;
 
 public class ConditionsSteps extends Steps {
-	
-	private ConditionsPage conditionsPage;
```
Comment (sherrif10) [inline src/test/java/org/openmrs/contrib/qaframework/automation/ConditionsSteps.java:29]: Seems you are changing the format from the former formatting , was it a recommendation ?

Comment (mherman22) [reply src/test/java/org/openmrs/contrib/qaframework/automation/ConditionsSteps.java:29]: Yes it was a recommendation to have the test create its own patient

**Comment by kdaud** [comment]
@mherman22 is this PR adding visibility element to Patient condition workflow test as the commit title starts?

**Comment by mherman22** [comment]
@kdaud refering to your question https://github.com/openmrs/openmrs-contrib-qaframework/pull/230?notification_referrer_id=NT_kwDOAs7_ibMyNjk5MDk4NDkzOjQ3MTIwMjY1&notifications_query=is%3Aunread#issuecomment-972778069 , i believe i did so. i changed the cssSelectors of the delete buttons and replaced them with the new selectors as per [30](https://github.com/openmrs/openmrs-contrib-qaframework/blob/fc79886464d38b0168b18abe3002d580acd6cee6/src/test/java/org/openmrs/contrib/qaframework/page/ConditionsPage.java#L30) and [31](https://github.com/openmrs/openmrs-contrib-qaframework/blob/fc79886464d38b0168b18abe3002d580acd6cee6/src/test/java/org/openmrs/contrib/qaframework/page/ConditionsPage.java#L31). I then added `waiter.until(ExpectedConditions.elementToBeClickable` to those elements which were not visible. i didnt use visibilityOfElementLocated because all it does is check whether the element is visible on the DOM. i used `elementToBeClickable` because it is an expectation for checking an element is visible and enabled such that you can click it.

referred to https://www.selenium.dev/selenium/docs/api/java/org/openqa/selenium/support/ui/ExpectedConditions.html

File: src/test/resources/features/refapp-2.x/12-condition/conditions.feature [hunk 3]
```diff
@@ -1,43 +1,66 @@
...
 
-  @dashboard
-  Scenario: Set first condition to inactive
+  @selenium
+  @condition
+  Scenario: Active condition management
+    # User story: Adding an active condition
+    And User enters active condition
+    And User clicks save
+
```
Comment (kdaud) [inline src/test/resources/features/refapp-2.x/12-condition/conditions.feature:15]: was it an oversight to leave out this empty line?
```suggestion
```

File: src/test/resources/features/refapp-2.x/12-condition/conditions.feature [hunk 4]
```diff
@@ -1,43 +1,66 @@
...
 
-  @dashboard
-  Scenario: Set first condition to active
+  @selenium
+  @condition
+  Scenario: Inactive condition management
+    # User story: Adding an active condition
+    And User enters inactive condition
+    And User clicks save
+
```
Comment (kdaud) [inline src/test/resources/features/refapp-2.x/12-condition/conditions.feature:26]:
```suggestion
```

File: src/test/resources/features/refapp-2.x/12-condition/conditions.feature [hunk 5]
```diff
@@ -1,43 +1,66 @@
...
     Then System should move condition to active section
 
-  @dashboard
-  Scenario: Edit first active condition
+  @selenium
+  @condition
+  Scenario: Editing First Active condition
+    And User enters active condition
+    And User clicks save
+
```
Comment (kdaud) [inline src/test/resources/features/refapp-2.x/12-condition/conditions.feature:36]:
```suggestion
```

File: src/test/resources/features/refapp-2.x/12-condition/conditions.feature [hunk 6]
```diff
@@ -1,43 +1,66 @@
...
     Then System should edit all active adjustable fields
 
-  @dashboard
-  Scenario: Edit first inactive condition
+  @selenium
+  @condition
+  Scenario: Editing First Inactive condition
+    And User enters inactive condition
+    And User clicks save
+
```
Comment (kdaud) [inline src/test/resources/features/refapp-2.x/12-condition/conditions.feature:46]:
```suggestion
```

File: src/test/java/org/openmrs/contrib/qaframework/automation/ConditionsSteps.java (hunk 1 above)
Comment (kdaud) [reply src/test/java/org/openmrs/contrib/qaframework/automation/ConditionsSteps.java:213]: @sherrif10 do you have any response on the caption [above](https://github.com/openmrs/openmrs-contrib-qaframework/pull/230/files#r751346120)?

Comment (sherrif10) [reply src/test/java/org/openmrs/contrib/qaframework/automation/ConditionsSteps.java:213]: @kdaud  am okay with that approach after inspecting it.

**Review by kdaud — APPROVED** [review]
@sherrif10 what do you say about this PR?

File: src/test/java/org/openmrs/contrib/qaframework/automation/ConditionsSteps.java [hunk 7]
```diff
@@ -25,27 +26,30 @@
...
+    }
+ 
+    @After(RunTest.HOOK.SELENIUM_CONDITION)
+    public void destroy() {
+        deletePatient(testPatient);
+        quit();
+    }
+    
 	@Given("User clicks on Conditions from Patient dashboard")
 	public void launchManageConditions() {
```
Comment (sherrif10) [inline src/test/java/org/openmrs/contrib/qaframework/automation/ConditionsSteps.java:51]: Fix the identation to be same . You have two factor identation format, cross check and confirm especially the before and after hook and other methods definition.

**Comment by sherrif10** [comment]
We can make the title as `RATEST-242: Adding Visibility Element to refapp-2.x Patient conditions workflow`

File: src/test/java/org/openmrs/contrib/qaframework/automation/ConditionsSteps.java (hunk 7 above)
Comment (mherman22) [reply src/test/java/org/openmrs/contrib/qaframework/automation/ConditionsSteps.java:51]: @sherrif10 i have fixed the indention in the commit https://github.com/openmrs/openmrs-contrib-qaframework/pull/230/commits/34fc2eba733c5a1ecf2e395ef33ec2d6cbd11cb0

**Review by sherrif10 — APPROVED** [review]
Thanks @mherman22   . LGTM  cc @dkayiwa  @kdaud this seems ripen for entering into master

**Pull request merged**